spec/requests/courses_spec.rb

```diff
@@ -1,19 +1,73 @@
 require 'helpers/rails_helper'
+require 'helpers/api_helper'
 
 RSpec.describe "Courses", type: :request do
   include_context "with authenticated requests"
-  let(:user) { create(:user) }
+  let(:student) { create(:user) }
+  let(:teacher) { create(:user, :teacher) }
   let!(:course) { create(:course) }
 
-  before do
-    allow(Auth).to receive(:decode).and_return({ 'user_id' => user.id })
-  end
-
   describe "GET /api/courses" do
-    it "returns the Course as JSON" do
+    it "returns the Course as JSON for students" do
+      authenticate(student)
+
       get "/api/courses/#{course.id}"
       expect(response).to have_http_status(200)
-      expect(response.body).to eq(course.to_json)
+      parsed = ActiveSupport::JSON.decode(response.body)
+      validate_course(parsed, course)
+    end
+  end
+
+  describe "GET /api/courses/visible" do
+    it "returns the Courses that the student is enrolled in as JSON" do
```

Comment:
Let's make this into two separate cases: one where the student is enrolled in something, and one where the student is not. Go for clarity of the test over DRY.